doc/releases/changelog-dev.md

```diff
@@ -3,6 +3,55 @@
 # Release 0.20.0-dev (development release)
 
 <h3>New features since last release</h3>
+* The `qml.fourier.reconstruct` function is added. It can be used to
+  reconstruct one-dimensional Fourier series with a minimal number of calls
+  to the original function.
+  [(#1864)](https://github.com/PennyLaneAI/pennylane/pull/1864)
+
+  The used reconstruction technique differs for functions with equidistant frequencies
+  that are reconstructed using the function value at equidistant sampling points and
+  for functions with arbitrary frequencies reconstructed using arbitrary sampling points.
+
+  As an example, consider the following QNode:
+
+  ```python
+  dev = qml.device("default.qubit", wires=2)
+  
+  @qml.qnode(dev)
+  def circuit(x, Y, f=1.0):
+      qml.RX(f*x, wires=0)
+      qml.RY(Y[0], wires=0)
+      qml.RY(Y[1], wires=1)
+      qml.CNOT(wires=[0, 1])
+      qml.RY(3*Y[1], wires=1)
+      return qml.expval(qml.PauliZ(0)@qml.PauliZ(1))
+  ```
+
+  It has three variational parameters overall: A scalar input `x`
+  and an array-valued input `Y` with two entries. Additionally, we can
+  tune the dependence on `x` with the frequency `f`.
+  We then can reconstruct the QNode output function with respect to `x` via
+
+  ```pycon
+  >>> x = 0.3
+  >>> Y = np.array([0.1, -0.9])
+  >>> rec = qml.fourier.reconstruct(circuit, ids="x", nums_frequency={"x": {0: 1}})(x, Y)
+  >>> rec
+  {'x': {0: <function pennylane.fourier.reconstruct._reconstruct_equ.<locals>._reconstruction(x)>}}
+  ```
+
+  As we can see, we get a nested dictionary in the format of the input `nums_frequency`
+  with functions as values. These functions are simple float-to-float callables:
+
+  ```pycon
+  >>> univariate = rec["x"][0]
+  >>> univariate(x)
+  -0.880208251507
+  ```
```

Comment:
1. It almost works :D JAX I could implement easily, for TF and Torch some more work needs to be done. I think it would be nice to have this, so I'll include it in this PR :)
2. The returned reconstructed functions are built using `qml.math`, so yes, they should be differentiable (not able to check for all frameworks because of 1.) Interestingly, depending on the combination of QNode interface and differentiation library for the reconstruction, they may differ: A QNode using the `autograd` interface can be reconstructed and afterwards one may call `jax.grad` on the reconstructed function, as long as the arguments given to the reconstruction/its gradient are `JAX`-compatible as well. :)
3. Similar to 2., the functions seems straightforwardly jittable, yes :)

I'll resolve the compatibility where necessary and then add some sentences in the changelog and documentation about this.